src/netpol_synth.py

```diff
@@ -154,33 +172,93 @@ def _xgress_conns_to_rules(conns, is_ingress):
 
         return res_rules
 
-    def synthesize(self, output_file):
-        """
-        Generates NetworkPolicies in yaml format based on the analysis done in the ctor.
-        If output file is specified, the output is dumped into the file. Otherwise, stdout is used
-        :param output_file: A file opened for writing
-        :return: None
-        """
+    def _find_deployments_from_pod_selector(self, selector_dict):
+        res = []
+        for deploy in self.deployments.values():
+            labels = deploy.labels
+            selector_labels = selector_dict['podSelector']['matchLabels']
```

Comment:
A `matchExpression` can also appear here (try changing `app=loadgenerator` to `app!=loadgenerator`). This will crash the code. Also check if IPBlocks can also appear here.
I suggest storing the actual Selector class in the connection and use its `matches` method here.

Reply:
added support for ip blocks + stored the actual Selector object for using its matches method